server/app/services/call_service.py

```
@@ -683,6 +695,23 @@ async def initiate_call(
+        # Validate rooftop VAPI configuration before proceeding
+        # Only validate when using rooftop's agent ID, not custom agent IDs
+        if rooftop and not custom_agent_id:
+            logger.debug(f"Validating VAPI configuration for rooftop: {rooftop.name}")
+            try:
+                validate_rooftop_configuration(
+                    agent_id=rooftop.agent_feedback_agent_id,
+                    phone_number_id=rooftop.feedback_agent_phone_number,
+                    rooftop_name=rooftop.name
+                )
+            except ValueError as e:
+                logger.error(f"VAPI configuration validation failed for rooftop {rooftop.name}: {str(e)}")
```

Ruff (0.14.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)